### PR TITLE
dhcp-server: T6079: Disable duplicate static-mappings on migration

### PR DESCRIPTION
**Pull request by sarthurdev**
## Change Summary

- Add migrator to disable duplicate static-mappings that would prevent config from loading moving to 1.4

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6079

## Related PR(s)

- https://github.com/vyos/vyos-1x/pull/3061

## Component(s) name

dhcp-server

## Proposed changes

- Add migrator to disable duplicate static-mappings
- Update basic-vyos configtest to validate migration

## How to test

```
vyos@vyos# load /usr/libexec/vyos/tests/config/basic-vyos
Loading configuration from '/usr/libexec/vyos/tests/config/basic-vyos'
Load complete. Use 'commit' to make changes effective.
[edit]
vyos@vyos# show service dhcp-server
+shared-network-name LAN {
+    authoritative
+    subnet 192.168.0.0/24 {
+        default-router 192.168.0.1
+        domain-name vyos.net
+        domain-search vyos.net
+        name-server 192.168.0.1
+        range LANDynamic {
+            start 192.168.0.20
+            stop 192.168.0.240
+        }
+        static-mapping TEST1 {
+            ip-address 192.168.0.21
+            mac-address 00:01:02:03:04:05
+        }
+        static-mapping TEST2 {
+            disable
+            ip-address 192.168.0.22
+            mac-address 00:01:02:03:04:05
+        }
+    }
+}
```

## Smoketest result

```
DEBUG - test_basic_vyos (__main__.TestConfigBasicVyos.test_basic_vyos) ...
DEBUG - WARNING: changing speed/duplex setting on "eth1" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth0" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth2" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth1" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth0" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth2" is unsupported!
DEBUG -
DEBUG -  time: 20.313
DEBUG - ok
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: smoketest/configs/basic-vyos [hunk 1]
```diff
@@ -87,6 +87,14 @@ service {
                     start 192.168.0.20
                     stop 192.168.0.240
                 }
+                static-mapping TEST1 {
+                    ip-address 192.168.0.21
+                    mac-address 00:01:02:03:04:05
+                }
+                static-mapping TEST2 {
```
Comment (c-po) [inline smoketest/configs/basic-vyos]: This is a testcase for duplicate MACs with different IPs, can we also have a case for duplicate IPs with different MACs?

**Review by c-po — CHANGES_REQUESTED** [review]

File: smoketest/configs/basic-vyos (hunk 1 above)
Comment (c-po) [reply smoketest/configs/basic-vyos]: Added in  0270932

**Review by c-po — APPROVED** [review]

**Comment by c-po** [comment]
```
DEBUG - vyos@vyos:~$ /usr/bin/vyos-configtest
DEBUG - /usr/bin/vyos-configtest
DEBUG - Generating tests
DEBUG - ... completed: 0.002374
DEBUG - test_basic_api_service (__main__.TestConfigBasicApiService.test_basic_api_service) ...
DEBUG - WARNING: No certificate specified, using build-in self-signed
DEBUG - certificates. Do not use them in a production environment!
DEBUG -
DEBUG -  time: 14.224
DEBUG - ok
DEBUG - test_basic_vyos (__main__.TestConfigBasicVyos.test_basic_vyos) ...
DEBUG - WARNING: changing speed/duplex setting on "eth1" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth0" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth2" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth1" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth0" is unsupported!
DEBUG -
DEBUG -
DEBUG - WARNING: changing speed/duplex setting on "eth2" is unsupported!
DEBUG -
DEBUG -  time: 21.619
DEBUG - ok
```

File: src/migration-scripts/dhcp-server/6-to-7 [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2021 VyOS maintainers and contributors
```
Comment (sever-sever) [inline src/migration-scripts/dhcp-server/6-to-7]: 2024 :)

Comment (c-po) [reply src/migration-scripts/dhcp-server/6-to-7]: Fixed

**Review by dmbaturin — APPROVED** [review]

**Pull request merged**